llvm/lib/CodeGen/GlobalISel/CallLowering.cpp

```diff
@@ -802,11 +847,16 @@ bool CallLowering::handleAssignments(ValueHandler &Handler,
         Register StackAddr = Handler.getStackAddress(
             MemTy.getSizeInBytes(), VA.getLocMemOffset(), MPO, Flags);
 
+        // Finish the handling of indirect passing from the passers
+        // (OutgoingParameterHandler) side.
+        // This branch is needed, so the pointer to the value is loaded onto the
+        // stack.
+        if (IndirectParameterPassingHandled) {
```

Comment:
Shouldn't it be `VA.getLocInfo() == CCValAssign::Indirect` so that we do the correct assignment for incoming arguments too? I guess this is why the memory operand has the wrong type in the test.

Reply:
I thought this was something we could do, but we need to know this information outside of the loop, and I think VA is out of scope.

Reply:
I am checking it.

Reply:
You are right.

Reply:
I thought you were suggesting to replace `IndirectParameterPassingHandled` with `VA.getLocInfo() == CCValAssign::Indirect`, but after reading again, I think you are suggesting something else. I thought that these two variables should have the same value, but based on your suggestion, maybe I misread the code.

Reply:
Also, it looks like StackAddr created just above should be different for Indirect case (MemTy should be replaced by PointerTy). It should affect the size of the stack object created for the pointer. (Unfortunately, stack objects are not verified by autogenerated checks.)

Reply:
> I thought you were suggesting to replace `IndirectParameterPassingHandled` with `VA.getLocInfo() == CCValAssign::Indirect`, but after reading again, I think you are suggesting something else.

The current version looks like what I had in mind. Please excuse my poor English.